### PR TITLE
feat(api): add commands to run standalone HTTP API

### PR DESCRIPTION
**Pull request by oXtxNt9U**
## Summary

Introduce a separate `api` package which contains commands to manage the new HTTP API and run it as a separate process.

To make this work, the `Application` class now supports two new flags to change the config loader behavior which are set implicitly in the respective run commands:

```
const flags: Contracts.AnyObject = {
	...this.getFlags(),
	initializationFileName: "api.json",
	allowMissingConfigFiles: true,
};
```

---

How to run the new API:

```
cd packages/api

# Publish
pnpm run api config:publish --token ark --network testnet --reset

# Run and listen on port 4003
pnpm run api api:run --env=test --token=ark --network=testnet
```

The API only needs access to Postgres.

NOTE:
The configs are currently published to the same directory that `packages/core` uses. Publishing the latter with `--reset` will delete the `api.json`. To workaround it, publish the api last.

## Checklist

- [ ] Documentation _(if necessary)_
- [ ] Tests _(if necessary)_
- [ ] Ready to be merged

### CODE REVIEW

File: packages/api/bin/config/testnet/api.json [hunk 1]
```diff
@@ -0,0 +1,76 @@
+{
+	"plugins": [
```
Comment (sebastijankuzner) [inline packages/api/bin/config/testnet/api.json:2]: List should contain only logger-pino, api-database, api-http.

File: packages/api/jest.config.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+module.exports = {
```
Comment (sebastijankuzner) [inline packages/api/jest.config.js:1]: We don't use jest anymore. File can be removed.

File: packages/api/package.json [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+		"build:watch": "pnpm run clean && tsc -w",
+		"clean": "del distribution",
+		"full:testnet": "cross-env CORE_PATH_CONFIG=./bin/config/testnet pnpm run mainsail core:run -- --env=test --token=ark --network=testnet",
+		"test": "uvu -r tsm source .test.ts",
+		"test:coverage": "c8 pnpm run test",
+		"test:coverage:html": "c8 -r html --all pnpm run test",
+		"test:file": "uvu -r tsm source"
+	},
+	"dependencies": {
+		"@mainsail/api-database": "workspace:*",
```
Comment (sebastijankuzner) [inline packages/api/package.json:28]: Remove unused dependencies.

File: packages/api/package.json [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+	"types": "distribution/index.d.ts",
+	"bin": {
+		"api": "./bin/run"
+	},
+	"files": [
+		"/bin",
+		"/distribution"
+	],
+	"scripts": {
+		"api": "node ./bin/run",
```
Comment (sebastijankuzner) [inline packages/api/package.json:17]: Rename command to mainsail-api.

File: packages/api/source/commands/config-publish.ts [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+	async #performPublishment(flags: Contracts.AnyObject): Promise<void> {
+		this.app
+			.rebind(Identifiers.ApplicationPaths)
+			.toConstantValue(this.environment.getPaths(flags.token, flags.network));
+
+		const configDestination = this.app.getCorePath("config");
+		const configSource = resolve(__dirname, `../../bin/config/${flags.network}`);
+
+		await this.components.taskList([
+			// TODO: overwrites core .env... so ideally we would use separate folders for core and api?
```
Comment (sebastijankuzner) [inline packages/api/source/commands/config-publish.ts:64]: Separate sub folders can be created for each process. Name them mainsail & mainsail-api to match the process names.

File: packages/api/source/exceptions/crypto.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import { Exception } from "./base";
```
Comment (sebastijankuzner) [inline packages/api/source/exceptions/crypto.ts:1]: Can we remove this file?

File: packages/api/source/commands/config-publish.ts (hunk 5 above)
Comment (oXtxNt9U) [reply packages/api/source/commands/config-publish.ts]: Done https://github.com/ArkEcosystem/mainsail/pull/231/commits/4e22c6b4c7d888e0f5701c8bffb3ee7b4771c347

**Pull request merged**